scom/stackstate_checks/scom/scom.py

```diff
@@ -122,16 +147,17 @@ def nextNodes(self, serviceid, session, scom_ip, domain, username, password, sys
                 ]
             })
             self.relation(str(serviceid), str(idd.get("id")), "hosted_on", {})
-            self.sendAlerts(session, scom_ip, domain, username, password, idd.get("id"))
+            if "api" in integration_mode:
+               self.sendAlerts(session, scom_ip, domain, username, password, idd.get("id"))
         return ids
 
-    def serviceTree(self, serviceidtree, session, scom_ip, domain, username, password, system_type):
+    def serviceTree(self, serviceidtree, session, scom_ip, domain, username, password, system_type,integration_mode):
```

Comment:
It is best to take all these variables `scom_ip, domain, username, password, system_type,integration_mode` and stick them in a single `ScomInstanceConfig` class. Then you can pass that around.